### PR TITLE
Add support for `function_to_apply` in classification

### PR DESCRIPTION
**Pull request by coderrg**
For classification models, users should have some way of specifying the function to be applied to the logits rather than `softmax` always being used since the logits do not always correspond to mutually exclusive classes.

Take, for example, a model's logits representing whether a piece of text conveys negative, neutral, or positive sentiment in a text classification task. In this case, the `softmax` activation function is apt as it ensures that the probabilities in the final output vector sum to 1. However, if the logits denote whether the text is toxic, obscene, insulting, etc. (as seen in [Detoxify models](https://github.com/unitaryai/detoxify)), the `sigmoid` activation function would be more appropriate since the values in the final output vector do not necessarily have to sum to 1.

HF pipelines used `softmax` as the mandatory activation function over multi-class model logits until https://github.com/huggingface/transformers/pull/8328, when they added support for the `function_to_apply` parameter which can be specified by the user.

I am proposing support for a `function_to_apply` parameter here as well. The options are the same as implemented in HF: `softmax`, `sigmoid`, and `none` (`none` applies no activation function and just returns the raw logits). `softmax` is the default since it preserves backward compatibility and seems to be the most common case.

### CODE REVIEW

File: lib/bumblebee/text/text_classification.ex [hunk 1]
```diff
@@ -24,7 +27,21 @@ defmodule Bumblebee.Text.TextClassification do
...
+
+        "sigmoid" ->
+          Axon.Activations.sigmoid(outputs.logits)
+
+        "none" ->
+          outputs.logits
+
+        _ ->
+          raise ArgumentError,
+                "Invalid :function_to_apply option. Only 'softmax', 'sigmoid', and 'none' are accepted"
```
Comment (jonatanklosko) [inline lib/bumblebee/text/text_classification.ex:43]:
```suggestion
        other ->
          raise ArgumentError,
                "expected :scores_function to be either of :softmax, :sigmoid or :none, got: #{inspect(other}"
```

File: lib/bumblebee/text/text_classification.ex [hunk 2]
```diff
@@ -24,7 +27,21 @@ defmodule Bumblebee.Text.TextClassification do
...
+      end
```
Comment (jonatanklosko) [inline lib/bumblebee/text/text_classification.ex:44]: Also, we can move this whole block to the `Shared`, can be `Shared.logits_to_scores(logits, scores_function)` :)

**Comment by coderrg** [comment]
Thank you for your feedback, @jonatanklosko! I have implemented your suggested changes.

File: lib/bumblebee/text/text_classification.ex [hunk 3]
```diff
@@ -6,10 +6,13 @@ defmodule Bumblebee.Text.TextClassification do
   def text_classification(model_info, tokenizer, opts \\ []) do
     %{model: model, params: params, spec: spec} = model_info
     Shared.validate_architecture!(spec, :for_sequence_classification)
-    opts = Keyword.validate!(opts, [:compile, top_k: 5, defn_options: []])
+
+    opts =
+      Keyword.validate!(opts, [:compile, top_k: 5, scores_function: "softmax", defn_options: []])
```
Comment (jonatanklosko) [inline lib/bumblebee/text/text_classification.ex:11]:
```suggestion
      Keyword.validate!(opts, [:compile, top_k: 5, scores_function: :softmax, defn_options: []])
```

Same for the rest :)

Comment (jonatanklosko) [reply lib/bumblebee/text/text_classification.ex:11]: And let's also document the new option in [respective places](https://github.com/elixir-nx/bumblebee/blob/776e57c6b6d06c0fed47afa26d8144c7c2541149/lib/bumblebee/text.ex#L257).

Comment (coderrg) [reply lib/bumblebee/text/text_classification.ex]: Thanks! Adjusted as per your suggestions.

**Review by jonatanklosko — APPROVED** [review]
Awesome, thanks :)
